### DISCUSSION TITLE
Error: Failure while executing; `cp -pR /private/tmp/d20220219-4070-1fgs9u3/glib/. /opt/homebrew/Cellar/glib` exited with 1.

### DISCUSSION THREAD

**Discussion by dbz2k** [Homebrew/discussions#2990] [Everyday usage] [answered]
`Error: Failure while executing; `cp -pR /private/tmp/d20220219-4070-1fgs9u3/glib/. /opt/homebrew/Cellar/glib` exited with 1. Here's the output:
cp: /opt/homebrew/Cellar/glib/./2.70.4/bin/gtester: Permission denied
cp: /opt/homebrew/Cellar/glib/./2.70.4/bin/gdbus-codegen: Permission denied
cp: /opt/homebrew/Cellar/glib/./2.70.4/bin/gsettings: Permission denied
cp: /opt/homebrew/Cellar/glib/./2.70.4/bin/glib-compile-schemas: Permission denied
cp: /opt/homebrew/Cellar/glib/./2.70.4/bin/gobject-query: Permission denied
cp: /opt/homebrew/Cellar/glib/./2.70.4/bin/gio-querymodules: Permission denied
cp: /opt/homebrew/Cellar/glib/./2.70.4/bin/glib-compile-resources: Permission denied
cp: /opt/homebrew/Cellar/glib/./2.70.4/bin/gresource: Permission denied
cp: /opt/homebrew/Cellar/glib/./2.70.4/bin/gtester-report: Permission denied
cp: /opt/homebrew/Cellar/glib/./2.70.4/bin/glib-gettextize: Permission denied
cp: /opt/homebrew/Cellar/glib/./2.70.4/bin/gdbus: Permission denied
cp: /opt/homebrew/Cellar/glib/./2.70.4/bin/glib-genmarshal: Permission denied
cp: /opt/homebrew/Cellar/glib/./2.70.4/bin/glib-mkenums: Permission denied
cp: /opt/homebrew/Cellar/glib/./2.70.4/bin/gio: Permission denied
cp: /opt/homebrew/Cellar/glib/./2.70.4/lib/libgmodule-2.0.a: Permission denied
cp: /opt/homebrew/Cellar/glib/./2.70.4/lib/libgobject-2.0.a: Permission denied
cp: /opt/homebrew/Cellar/glib/./2.70.4/lib/pkgconfig/gio-2.0.pc: Permission denied
cp: /opt/homebrew/Cellar/glib/./2.70.4/lib/pkgconfig/gio-unix-2.0.pc: Permission denied
cp: /opt/homebrew/Cellar/glib/./2.70.4/lib/pkgconfig/gobject-2.0.pc: Permission denied
cp: /opt/homebrew/Cellar/glib/./2.70.4/lib/pkgconfig/glib-2.0.pc: Permission denied
cp: /opt/homebrew/Cellar/glib/./2.70.4/lib/pkgconfig/gmodule-export-2.0.pc: Permission denied
cp: /opt/homebrew/Cellar/glib/./2.70.4/lib/pkgconfig/gthread-2.0.pc: Permission denied
cp: /opt/homebrew/Cellar/glib/./2.70.4/lib/pkgconfig/gmodule-no-export-2.0.pc: Permission denied
cp: /opt/homebrew/Cellar/glib/./2.70.4/lib/pkgconfig/gmodule-2.0.pc: Permission denied
cp: /opt/homebrew/Cellar/glib/./2.70.4/lib/libglib-2.0.a: Permission denied
cp: /opt/homebrew/Cellar/glib/./2.70.4/lib/libgio-2.0.a: Permission denied
cp: /opt/homebrew/Cellar/glib/./2.70.4/lib/libgthread-2.0.a: Permission denied
cp: /opt/homebrew/Cellar/glib/./2.70.4/lib/glib-2.0/include/glibconfig.h: Permission denied
`

Stuff like this keeps happening when installing updating bottles I have to fix it by
1. brew doctor then do the "brew link" on the package
2. brew reinstall the package that causes the issue

How would I fix this.

**Comment by carlocab** [comment]
What is the output of `brew config` and `brew doctor`?

**Reply by dbz2k** [reply]
Brew config

`HOMEBREW_VERSION: 3.3.15
ORIGIN: https://github.com/Homebrew/brew
HEAD: d603fd6426ddae27327bb51afb085653eb2d84a7
Last commit: 5 days ago
Core tap: N/A
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_INSTALL_FROM_API: set
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.8 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 13.0.0 build 1300
Git: 2.35.1 => /opt/homebrew/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.6.3-arm64
CLT: 13.2.0.0.1.1638488800
Xcode: N/A
Rosetta 2: false
`


Brew doctor
`Your system is ready to brew.`

**Reply by carlocab** [reply]
This looks wrong:
```
Core tap: N/A
```
It's weird that `brew doctor` isn't complaining about it, though.

Try
```
rm -fr /opt/homebrew/Library/Taps/homebrew/homebrew-core
brew tap homebrew/core
brew update-reset
```

When you've done that, please show the output of `brew config` and `brew doctor` again.

**Comment by dbz2k** [comment, marked as answer]
Been testing this for a while for some reason "HOMEBREW_INSTALL_FROM_API" caused  this permission error somehow. I disable it, and did "brew update-reset" had no problem since

**Reply by ShadowBullet25** [reply]
so, how would you go about disabling this? tried google, but i didnt get anything that i could understand

**Reply by viasux** [reply]
I'm having this issue as well and would like to know how to disable HOMEBREW_INSTALL_FROM_API

**Reply by SMillerDev** [reply]
Don't set it

**Reply by viasux** [reply]
oh, I never did set it, to my knowledge - although another person has used this computer in the past and may have done something with brew that I'm unaware of. It would have been on their user though.
